src/Util/ParseUrl.php

```diff
@@ -161,16 +161,14 @@ public static function getSiteinfo($url, $no_guessing = false, $do_oembed = true
 					$siteinfo['type'] = $oembed_data->type;
 				}
 
-				if (($oembed_data->type == 'link') && ($siteinfo['type'] != 'photo')) {
-					if (isset($oembed_data->title)) {
-						$siteinfo['title'] = trim($oembed_data->title);
-					}
-					if (isset($oembed_data->description)) {
-						$siteinfo['text'] = trim($oembed_data->description);
-					}
-					if (isset($oembed_data->thumbnail_url)) {
-						$siteinfo['image'] = $oembed_data->thumbnail_url;
-					}
+				if (isset($oembed_data->title)) {
```

Comment:
Why had you removed the check for ```($siteinfo['type'] != 'photo')```? This could easily lead to a problem with our picture posts (and yes: We need a better way to detect the types of our own posts).

Reply:
OEmbed already is structured info, if it's present, we should use it instead of guessing from the DOM document. If you have a comprehensive list of particular cases, I'm all ears. In the current state I don't see why we should do an exception.

Reply:
When we are posting to a connector network, we are guessing the site content type to decide what to post (text, photo, ...). Currently this is done much too complicated (by using this function). When something that had been detected as "photo" before is now detected as "link", then we won't post pictures to external networks anymore, but only links to the pictures.

We really should change this way of detecting the content type - but not in this version. So I really would made an exception for "photo".